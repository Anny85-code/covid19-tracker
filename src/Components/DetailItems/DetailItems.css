.country-detail {
  height: 100px;
  padding: 10px;
  display: flex;
  justify-content: space-between;
}

.hd {
  margin-right: 5px;
}

small {
  margin-right: 20px;
  font-size: 17px;
}

span {
  font-size: 17px;
  font-weight: bold;
}

.h {
  padding: 0;
}

.h > small {
  margin-right: 5px;
  font-size: 17px;
}

.h > span {
  font-weight: 100;
}

.country-detail:nth-child(odd) {
  background-color: hsl(342, 96%, 68%);
}

.display-items {
  background-color: hsl(342, 98%, 63%);
  height: 170vh;
  font-family: 'Lato', sans-serif;
}

.details-header {
  height: 200px;
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  font-family: 'Poppins', sans-serif;
}

li {
  list-style-type: none;
}

.country-map {
  padding-left: 10px;
}
