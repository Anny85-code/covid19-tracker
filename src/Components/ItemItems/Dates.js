const date = () => {
  const currentDate = new Date();
  const day =
    currentDate.getDate().toString().length === 1
      ? `0${currentDate.getDate().toString()}`
      : currentDate.getDate().toString();
  const year = currentDate.getFullYear().toString();
  const month =
    (currentDate.getMonth() + 1).toString().length === 1
      ? `0${(currentDate.getMonth() + 1).toString()}`
      : (currentDate.getMonth() + 1).toString();
  return `${year}-${month}-${day}`;
}

export default date;