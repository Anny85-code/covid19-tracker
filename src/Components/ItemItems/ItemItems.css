.li-content {
  border: 1px solid gray;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  color: #fff;
  list-style-type: none;
  padding-right: 15px;
  font-weight: bolder;
  font-family: 'Poppins', sans-serif;
  font-size: 18px;
}

.li-content:nth-child(even) {
  background-color: hsl(342, 54%, 44%);
}

.li-content:nth-child(odd) {
  background-color: hsl(342, 96%, 68%);
}

a {
  text-decoration: none;
}

.infos-circle {
  flex: 1;
}

.country-map {
  height: 150px;
  margin-right: 100px;
}

@media screen and (max-width: 600px) {
  .country-map {
    margin-right: 0;
  }

  .li-content {
    width: 170px;
    margin-right: 0;
    height: 270px;
  }

  ul {
    margin: 0;
    padding: 0;
  }
}
