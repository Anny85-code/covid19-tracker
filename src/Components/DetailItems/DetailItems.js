import PropTypes from 'prop-types';
import { BsArrowRightCircle } from 'react-icons/bs';
import './DetailItems.css';

const DetailItems = ({
  country,
  source,
  todayConfirmed,
  todayDeaths,
  todayNewConfirmed,
  todayNewDeaths,
  todayNewRecovered,
  yesterdayDeaths,
  todayRecovered,
  map,
}) => (
  <div className="display-items">
    <div className="details-header">
      <img className="country-map" alt="country-map" src={map} />
      <h2>{country}</h2>
    </div>
    <ul className="details-wrapper">
      <li className="country-detail">
        <span>Total Confirmed</span>
        <small>
          {todayConfirmed}
          <BsArrowRightCircle className="infos-circle" />
        </small>
      </li>
      <li className="source">
        <span>Source</span>
        <small>
          {source}
          <BsArrowRightCircle className="infos-circle" />
        </small>
      </li>
      <li className="country-detail">
        <span>Total Deaths</span>
        <small>
          {todayDeaths}
          <BsArrowRightCircle className="infos-circle" />
        </small>
      </li>
      <li className="country-detail">
        <span>Yesterday&apos;s Deaths</span>
        <small>
          {yesterdayDeaths}
          <BsArrowRightCircle className="infos-circle" />
        </small>
      </li>

      <li className="country-detail">
        <span>Today Recovered</span>
        <small>
          {todayRecovered}
          <BsArrowRightCircle className="infos-circle" />
        </small>
      </li>
      <li className="country-detail">
        <span>Today New Confirmed</span>
        <small>
          {todayNewConfirmed}
          <BsArrowRightCircle className="infos-circle" />
        </small>
      </li>
      <li className="country-detail">
        <span>Today New Deaths</span>
        <small>
          {todayNewDeaths}
          <BsArrowRightCircle className="infos-circle" />
        </small>
      </li>
      <li className="country-detail">
        <span>Today New Recovered</span>
        <small>
          {todayNewRecovered}
          <BsArrowRightCircle className="infos-circle" />
        </small>
      </li>
    </ul>
  </div>
);

DetailItems.propTypes = {
  todayConfirmed: PropTypes.number.isRequired,
  todayDeaths: PropTypes.number.isRequired,
  todayNewConfirmed: PropTypes.number.isRequired,
  todayNewDeaths: PropTypes.number.isRequired,
  source: PropTypes.string.isRequired,
  todayNewRecovered: PropTypes.number.isRequired,
  yesterdayDeaths: PropTypes.number.isRequired,
  todayRecovered: PropTypes.number.isRequired,
  country: PropTypes.string.isRequired,
  map: PropTypes.string.isRequired,
};
export default DetailItems;
