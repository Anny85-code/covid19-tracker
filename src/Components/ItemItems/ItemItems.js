import './ItemItems.css';
import propTypes from 'prop-types';
import { BsArrowRightCircle } from 'react-icons/bs';

const ItemItems = ({
  id, name, todayConfirmed, map,
}) => (
  <ul key={id} className="li-content">
    <li>
      <BsArrowRightCircle className="infos-circle" />
    </li>
    <li>
      <img className="country-map" alt="country-map" src={map} />
    </li>
    <li>
      <div className="content name">{name}</div>
      <div className="content status center-container">
        {new Intl.NumberFormat().format(todayConfirmed)}
      </div>
    </li>
  </ul>
);
ItemItems.propTypes = {
  id: propTypes.string.isRequired,
  name: propTypes.string.isRequired,
  todayConfirmed: propTypes.number.isRequired,
  map: propTypes.string.isRequired,
};

export default ItemItems;
