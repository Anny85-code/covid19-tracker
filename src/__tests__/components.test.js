import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import DetailItems from '../Components/DetailItems/DetailItems';
import ItemItems from '../Components/ItemItems/ItemItems';
import Loader from '../Components/Spinner/Spinner';

afterEach(cleanup);

describe('Snapshot of covid19 data', () => {
  it('Snapshot of ItemItems Component', () => {
    const tree = renderer.create(<ItemItems />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Snapshot of DetailItems Component', () => {
    const tree = renderer.create(<DetailItems />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Snapshot of Spinner Component', () => {
    const tree = renderer.create(<Loader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
