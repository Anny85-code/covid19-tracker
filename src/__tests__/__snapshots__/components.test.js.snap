// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of covid19 data Snapshot of DetailItems Component 1`] = `
<div
  className="display-items"
>
  <div
    className="details-header"
  >
    <img
      alt="country-map"
      className="country-map"
    />
    <div>
      <h2 />
      <li
        className="country-detail h"
      >
        <small />
        <span
          className="hd"
        >
          Total Confirmed
        </span>
      </li>
    </div>
  </div>
  <div
    className="stat"
  >
    Stat Breakdown
  </div>
  <ul
    className="details-wrapper"
  >
    <li
      className="source country-detail"
    >
      <span>
        Source
      </span>
      <span>
        <small />
        <small>
          <svg
            className="infos-circle"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              fillRule="evenodd"
            />
          </svg>
        </small>
      </span>
    </li>
    <li
      className="country-detail"
    >
      <span>
        Total Deaths
      </span>
      <span>
        <small />
        <small>
          <svg
            className="infos-circle"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              fillRule="evenodd"
            />
          </svg>
        </small>
      </span>
    </li>
    <li
      className="country-detail"
    >
      <span>
        Yesterday's Deaths
      </span>
      <span>
        <small />
        <small>
           
          <svg
            className="infos-circle"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              fillRule="evenodd"
            />
          </svg>
        </small>
      </span>
    </li>
    <li
      className="country-detail"
    >
      <span>
        Today Recovered
      </span>
      <span>
        <small />
        <small>
          <svg
            className="infos-circle"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              fillRule="evenodd"
            />
          </svg>
        </small>
      </span>
    </li>
    <li
      className="country-detail"
    >
      <span>
        Today New Confirmed
      </span>
      <span>
        <small />
        <small>
          <svg
            className="infos-circle"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              fillRule="evenodd"
            />
          </svg>
        </small>
      </span>
    </li>
    <li
      className="country-detail"
    >
      <span>
        Today New Deaths
      </span>
      <span>
        <small />
        <small>
          <svg
            className="infos-circle"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              fillRule="evenodd"
            />
          </svg>
        </small>
      </span>
    </li>
    <li
      className="country-detail"
    >
      <span>
        Today New Recovered
      </span>
      <span>
        <small />
        <small>
          <svg
            className="infos-circle"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              fillRule="evenodd"
            />
          </svg>
        </small>
      </span>
    </li>
  </ul>
</div>
`;

exports[`Snapshot of covid19 data Snapshot of ItemItems Component 1`] = `
<ul
  className="li-content"
>
  <li>
    <svg
      className="infos-circle"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
        fillRule="evenodd"
      />
    </svg>
  </li>
  <li>
    <img
      alt="country-map"
      className="country-map"
    />
  </li>
  <li>
    <div
      className="content name"
    />
    <div
      className="content status center-container"
    >
      NaN
    </div>
  </li>
</ul>
`;

exports[`Snapshot of covid19 data Snapshot of Spinner Component 1`] = `
<div
  className="spinner"
>
  <span
    className="css-1xdhyk6"
  >
    <span
      className="css-1ionjmu"
    />
    <span
      className="css-1dx4to7"
    />
    <span
      className="css-8elkgk"
    />
    <span
      className="css-1ip0fzg"
    />
    <span
      className="css-of60g4"
    />
  </span>
</div>
`;
